### PR TITLE
[FLINK-10166] [table] Reduce dependencies by removing org.apache.commons

### PR DESCRIPTION
**Pull request by twalthr**
## What is the purpose of the change

This PR removes all dependencies to `org.apache.commons` libraries. In the past we only used a couple of methods that were partially pulled in from Hadoop causing the issues mentioned in the JIRA ticket.

## Brief change log

- Add more utility method to the Flink code base in order to reduce external dependencies

## Verifying this change

This change is already covered by existing tests.
I added additional test such as `EncodingUtilsTest` and `TypeCheckUtilsTest`.

## Does this pull request potentially affect one of the following parts:

  - Dependencies (does it add or upgrade a dependency): yes
  - The public API, i.e., is any changed class annotated with `@Public(Evolving)`: no
  - The serializers: no
  - The runtime per-record code paths (performance sensitive): no
  - Anything that affects deployment or recovery: JobManager (and its components), Checkpointing, Yarn/Mesos, ZooKeeper: no
  - The S3 file system connector: no

## Documentation

  - Does this pull request introduce a new feature? no
  - If yes, how is the feature documented? not applicable

### CODE REVIEW

File: flink-libraries/flink-table-common/src/main/java/org/apache/flink/table/utils/EncodingUtils.java [hunk 1]
```diff
@@ -87,10 +95,145 @@ public static String encodeObjectToString(Serializable obj) {
...
+				final char ch0 = str.charAt(0);
+				final char ch1 = str.charAt(1);
+				final char[] output2 = new char[outputLength];
+				for (int i = repeat * 2 - 2; i >= 0; i--, i--) {
+					output2[i] = ch0;
+					output2[i + 1] = ch1;
+				}
+				return new String(output2);
+			default:
+				final StringBuilder buf = new StringBuilder(outputLength);
```
Comment (igalshilman) [inline flink-libraries/flink-table-common/src/main/java/org/apache/flink/table/utils/EncodingUtils.java]: I know that you've just copied it, but honestly, I think that the default case is the only one that makes senes and all the complication in that method is a bit wired (as the comment above states - it is optimized for jdk1.4) 
Feel free to ignore my comment, if you feel that fixing this, is out side of the scope of this PR :-)

Comment (twalthr) [reply flink-libraries/flink-table-common/src/main/java/org/apache/flink/table/utils/EncodingUtils.java]: If I change this method, I would also need to introduce more extensive tests in order to verify that I did not introduce new bugs. This method can be remove soon anyway as Java 11 has a `String.repeat` method.

Comment (igalshilman) [reply flink-libraries/flink-table-common/src/main/java/org/apache/flink/table/utils/EncodingUtils.java]: Got it. makes senes.

File: flink-libraries/flink-table-common/src/main/java/org/apache/flink/table/utils/EncodingUtils.java [hunk 2]
```diff
@@ -40,14 +44,18 @@
 
 	private static final Base64.Decoder BASE64_DECODER = java.util.Base64.getUrlDecoder();
 
+	private static final MessageDigest MD5_MESSAGE_DIGEST = getMd5MessageDigest();
```
[the remark this answers is not in the document]
Comment (igalshilman) [reply flink-libraries/flink-table-common/src/main/java/org/apache/flink/table/utils/EncodingUtils.java]: Thanks for addressing this.

**Comment by igalshilman** [comment]
LGTM 👍

**Comment by twalthr** [comment]
Thank you @igalshilman. I will merge this...

**Pull request closed without merging**